Modify RPC isFront function to match criteria with will be modified RPC geometries for Run, Run 3 and Run 4

eigen1907 opened the pull request:
#### PR description:

This PR modifies the isFront function inside the RPC Endcap Geometry Builder.
It will be applied in synchronization with PR (#47134) by Sunanda Banerjee.

The isFront function classifies RPCs as front or back chambers by RPC Ids.

As the PR (#47134) modifies the z coordinates of RPCs inside RPC endcap stations (RE-4, RE-3, RE-2, RE+2, RE+3, RE+4),
The criteria for the isFront function needs to be changed as well.
Therefore, this PR modifies the criteria for the isFront function correctly as the z coordinate changes.

For history and more details, see the previous meeting materials.
(https://indico.cern.ch/event/1475698/contributions/6214667/attachments/2962024/5210467/RPCGeomReport_241106RPCDPG_JShin.pdf)


#### PR validation:

To avoid errors, this PR should be used with #47134 and the new scenario.
Otherwise (when applying this PR alone), the error below will be raised by the selfTest function in the Geometry Builder.

Error reproduction)
```
A fatal system signal has occurred: abort signal
The following is the call stack containing the origin of the signal.
...
#9  0x00007f62d8e9276c in MuRingForwardDoubleLayer::selfTest() const () from /eos/home-j/joshin/workspace-eos/rpc-geometry/update-isfront/CMSSW_15_0_0_pre1/lib/el9_amd64_gcc12/libRecoMuonDetLayers.so
#10 0x00007f62d8e9359b in MuRingForwardDoubleLayer::MuRingForwardDoubleLayer(std::vector<ForwardDetRing const*, std::allocator<ForwardDetRing const*> > const&, std::vector<ForwardDetRing const*, std::allocator<ForwardDetRing const*> > const&) () from /eos/home-j/joshin/workspace-eos/rpc-geometry/update-isfront/CMSSW_15_0_0_pre1/lib/el9_amd64_gcc12/libRecoMuonDetLayers.so
#11 0x00007f62d8e9f36f in MuonRPCDetLayerGeometryBuilder::buildLayer(int, std::vector<int, std::allocator<int> > const&, int, int, std::vector<int, std::allocator<int> >&, RPCGeometry const&) () from /eos/home-j/joshin/workspace-eos/rpc-geometry/update-isfront/CMSSW_15_0_0_pre1/lib/el9_amd64_gcc12/libRecoMuonDetLayers.so
#12 0x00007f62d8e9fea8 in MuonRPCDetLayerGeometryBuilder::buildEndcapLayers(RPCGeometry const&) () from /eos/home-j/joshin/workspace-eos/rpc-geometry/update-isfront/CMSSW_15_0_0_pre1/lib/el9_amd64_gcc12/libRecoMuonDetLayers.so
#13 0x00007f62bb85a0bf in MuonDetLayerGeometryESProducer::produce(MuonRecoGeometryRecord const&) () from /cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_15_0_0_pre1/lib/el9_amd64_gcc12/pluginRecoMuonDetLayersPlugins.so
...
```

This change will also require a new payload for RPC Geometry when working with Global Tags.


#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

eigen1907 committed:
Modify RPC isFront function to align with modified RPC Geometry for Run, Run 3 and Run 4

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"185af904401064400d913724e9cf0316c675852a":{"files":["RecoMuon/DetLayers/src/MuonRPCDetLayerGeometryBuilder.cc"],"squashed":false,"time":1737351815}},"emoji":{},"last_seen_sha":"185af904401064400d913724e9cf0316c675852a","signatures":{}} -->

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-47138/43365

cmsbuild commented:
A new Pull Request was created by  @eigen1907 for master.

It involves the following packages:

- RecoMuon/DetLayers (**reconstruction**)


@cmsbuild, @jfernan2, @mandrenguyen can you please review it and eventually sign? Thanks.
@CeliaFernandez, @Fedespring, @HuguesBrun, @abbiendi, @andrea21z, @bellan, @cericeci, @jhgoh, @missirol, @rociovilar, @trocino this is something you requested to watch as well.
@antoniovilela, @mandrenguyen, @rappoccio, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

eigen1907 commented:
@cmsbuild Please test

eigen1907 commented:
@cmsbuild please test

perrotta commented:
please test
(#47134 is already merged in the IBs)

bsunanda commented:
The new scenarios are in #47158 (Run2), #47159 (Run3) and #47160 (Run4).

perrotta commented:
please abort

perrotta commented:
test parameters:
- pull_requests =  #47158, #47159, #47160

perrotta commented:
please test

cmsbuild commented:
-1

**Failed Tests**: UnitTests RelVals RelVals-INPUT AddOn 
**Size**: This PR adds an extra 16KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/summary.html
**COMMIT**: 185af904401064400d913724e9cf0316c675852a
**CMSSW**: CMSSW_15_0_X_2025-01-21-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/47138/43910/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 2 errors in the following unit tests:

<pre>---> test test_MC_22_crosscheck had ERRORS
---> test test_MC_23_crosscheck had ERRORS
</pre>

## RelVals

- [135.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/135.4_ZEEFS_13)```A fatal system signal has occurred: abort signal```
- [1306.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/1306.0_SingleMuPt1_UP15)```A fatal system signal has occurred: abort signal```
- [7.3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/7.3_CosmicsSPLoose2018)```A fatal system signal has occurred: abort signal```
<details>
<summary>Expand to see more relval errors ...</summary>

- [140.045](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/140.045_RunJetHT2022C)
- [10224.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/10224.0_TTbar_13+2017PU)
- [4.53](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/4.53_RunPhoton2012B)
- [5.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/5.1_TTbarFS)
- [312.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/312.0_Pyquen_ZeemumuJets_pt10_2760GeV_2022)
- [145.713](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/145.713_RunTau2024I)
- [145.301](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/145.301_RunDisplacedJet2024E)
- [13234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/13234.0_TTbar_14TeV+2022FS)
- [1000.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/1000.0_RunMinBias2011A)
- [16834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/16834.0_TTbar_14TeV+2025)
- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/11634.0_TTbar_14TeV+2022)
- [136.874](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/136.874_RunEGamma2018C)
- [9.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/9.0_Higgs200ChargedTaus)
- [145.202](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/145.202_RunEGamma02024D)
- [14234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/14234.0_TTbar_14TeV+2023FSPU)
- [8.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/8.0_BeamHalo)
- [136.731](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/136.731_RunSinglePh2016B)
- [13034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/13034.0_TTbar_14TeV+2024PU)
- [136.793](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/136.793_RunDoubleEG2017C)
- [141.042](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/141.042_RunZeroBias2023D)
- [12846.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/12846.0_ZEE_14+2024)
- [4.22](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/4.22_RunCosmics2011A)
- [25.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/25.0_TTbar)
- [25202.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/25202.0_TTbar_13)
- [145.408](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/145.408_RunPark2MuonLowMass02024F)
- [1330.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/1330.0_ZMM_13)
- [250202.181](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/250202.181_TTbar13TeVPUppmx2018)
- [145.014](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/145.014_RunZeroBias2024B)
- [14034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/14034.0_TTbar_14TeV+2023FS)
- [145.104](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/145.104_RunJetMET02024C)
- [139.001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/139.001_RunMinimumBias2021)
- [140.56](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/140.56_RunHI2018)
- [1001.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/1001.0_RunMinBias2011A)
- [12834.7](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/12834.7_TTbar_14TeV+2024_trackingMkFit)
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/12434.0_TTbar_14TeV+2023)
- [145.5](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/145.5_RunBTagMu2024G)
- [12834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/12834.0_TTbar_14TeV+2024)
- [145.604](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrix-results/145.604_RunJetMET02024H)
</details>



## RelVals-INPUT

- [145.711](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.711_RunParkingSingleMuon02024I)```145.711_RunParkingSingleMuon02024I/step2_RunParkingSingleMuon02024I.log```
- [1030.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/1030.0_RunHLTPhy2017B)```1030.0_RunHLTPhy2017B/step2_RunHLTPhy2017B.log```
- [145.7](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.7_RunBTagMu2024I)```145.7_RunBTagMu2024I/step2_RunBTagMu2024I.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [145.31](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.31_RunParkingLLP2024E)
- [1052.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/1052.0_RunCosmics2022F)
- [1052.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/1052.1_RunSiPixelCalCosmics2022F)
- [145.114](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.114_RunZeroBias2024C)
- [145.109](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.109_RunParkingHH2024C)
- [142.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/142.0_RunHIPhysicsRawPrime2023A)
- [145.209](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.209_RunParkingHH2024D)
- [145.611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.611_RunParkingSingleMuon02024H)
- [145.01](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.01_RunParkingLLP2024B)
- [145.612](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.612_RunParkingVBF02024H)
- [145.11](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.11_RunParkingLLP2024C)
- [145.409](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.409_RunParkingHH2024F)
- [145.408](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.408_RunPark2MuonLowMass02024F)
- [145.613](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.613_RunTau2024H)
- [145.401](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.401_RunDisplacedJet2024F)
- [145.614](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.614_RunZeroBias2024H)
- [145.403](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.403_RunHcalNZS2024F)
- [145.402](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.402_RunEGamma02024F)
- [145.405](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.405_RunMuon02024F)
- [145.404](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.404_RunJetMET02024F)
- [145.407](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.407_RunNoBPTX2024F)
- [145.406](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.406_RunMuonEG2024F)
- [144.6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/144.6_MinimumBias2010A)
- [1002.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/1002.0_RRD)
- [1002.3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/1002.3_RunZeroBias2022B)
- [1002.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/1002.4_RunDoubleMuon2022B)
- [138.5](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/138.5_ExpressCollisions2021)
- [138.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/138.4_PromptCollisions2021)
- [138.2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/138.2_ExpressCosmics2021)
- [138.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/138.1_PromptCosmics2021)
- [142.903](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/142.903_RunUPC2016PA)
- [142.902](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/142.902_RunUPC2024)
- [142.901](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/142.901_RunUPC2024)
- [145.414](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.414_RunZeroBias2024F)
- [145.412](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.412_RunParkingVBF02024F)
- [145.413](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.413_RunTau2024F)
- [136.897](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/136.897_RunCosmics2021CRUZET)
- [136.899](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/136.899_RunCosmics2021CRAFT)
- [1000.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/1000.0_RunMinBias2011A)
- [145.714](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.714_RunZeroBias2024I)
- [145.009](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.009_RunParkingHH2024B)
- [145.008](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.008_RunPark2MuonLowMass02024B)
- [145.713](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.713_RunTau2024I)
- [145.712](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.712_RunParkingVBF02024I)
- [145.005](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.005_RunMuon02024B)
- [145.004](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.004_RunJetMET02024B)
- [145.007](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.007_RunNoBPTX2024B)
- [145.006](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.006_RunMuonEG2024B)
- [145.001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.001_RunDisplacedJet2024B)
- [145.609](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.609_RunParkingHH2024H)
- [145.003](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.003_RunHcalNZS2024B)
- [145.002](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.002_RunEGamma02024B)
- [145.61](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.61_RunParkingLLP2024H)
- [145.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.1_RunBTagMu2024C)
- [1040.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/1040.0_RunZeroBias2017F)
- [2500.242](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/2500.242_muDPGNANO140Xrun3)
- [134.813](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/134.813_RunCosmics2015C)
- [145.702](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.702_RunEGamma02024I)
- [145.703](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.703_RunHcalNZS2024I)
- [145.701](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.701_RunDisplacedJet2024I)
- [145.706](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.706_RunMuonEG2024I)
- [145.707](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.707_RunNoBPTX2024I)
- [145.704](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.704_RunJetMET02024I)
- [145.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.0_RunBTagMu2024B)
- [145.012](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.012_RunParkingVBF02024B)
- [145.013](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.013_RunTau2024B)
- [145.708](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.708_RunPark2MuonLowMass02024I)
- [145.011](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.011_RunParkingSingleMuon02024B)
- [145.014](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.014_RunZeroBias2024B)
- [140.57](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/140.57_RunHI2018Reduced)
- [145.311](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.311_RunParkingSingleMuon02024E)
- [145.313](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.313_RunTau2024E)
- [145.312](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.312_RunParkingVBF02024E)
- [145.314](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.314_RunZeroBias2024E)
- [145.71](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.71_RunParkingLLP2024I)
- [134.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/134.0_MinimumBiasBS)
- [145.306](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.306_RunMuonEG2024E)
- [145.307](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.307_RunNoBPTX2024E)
- [145.304](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.304_RunJetMET02024E)
- [145.305](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.305_RunMuon02024E)
- [145.302](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.302_RunEGamma02024E)
- [145.303](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.303_RunHcalNZS2024E)
- [145.301](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.301_RunDisplacedJet2024E)
- [145.308](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.308_RunPark2MuonLowMass02024E)
- [145.309](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.309_RunParkingHH2024E)
- [140.61](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/140.61_RunHI2022)
- [136.733](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/136.733_RunCosmics2016B)
- [145.41](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.41_RunParkingLLP2024F)
- [145.705](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.705_RunMuon02024I)
- [141.901](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/141.901_RunUPC2023)
- [141.902](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/141.902_RunUPC2023)
- [145.608](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.608_RunPark2MuonLowMass02024H)
- [143.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/143.911_RunUPC2024)
- [145.603](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.603_RunHcalNZS2024H)
- [145.602](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.602_RunEGamma02024H)
- [145.601](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.601_RunDisplacedJet2024H)
- [145.607](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.607_RunNoBPTX2024H)
- [145.606](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.606_RunMuonEG2024H)
- [145.605](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.605_RunMuon02024H)
- [145.604](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.604_RunJetMET02024H)
- [145.709](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.709_RunParkingHH2024I)
- [145.51](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.51_RunParkingLLP2024G)
- [145.411](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.411_RunParkingSingleMuon02024F)
- [145.513](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.513_RunTau2024G)
- [145.512](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.512_RunParkingVBF02024G)
- [145.511](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.511_RunParkingSingleMuon02024G)
- [145.514](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.514_RunZeroBias2024G)
- [4.6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/4.6_MinimumBias2010A)
- [145.106](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.106_RunMuonEG2024C)
- [143.902](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/143.902_RunUPC2024)
- [1001.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/1001.4_RunDoubleMuon2022B)
- [143.901](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/143.901_RunUPC2024)
- [145.113](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.113_RunTau2024C)
- [145.112](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.112_RunParkingVBF02024C)
- [145.111](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.111_RunParkingSingleMuon02024C)
- [145.208](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.208_RunPark2MuonLowMass02024D)
- [145.207](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.207_RunNoBPTX2024D)
- [145.206](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.206_RunMuonEG2024D)
- [145.205](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.205_RunMuon02024D)
- [145.204](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.204_RunJetMET02024D)
- [145.203](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.203_RunHcalNZS2024D)
- [145.202](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.202_RunEGamma02024D)
- [145.201](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.201_RunDisplacedJet2024D)
- [1001.3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/1001.3_RunSingleMuon2022B)
- [1001.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/1001.0_RunMinBias2011A)
- [1005.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/1005.0_RunCosmics2022F)
- [145.501](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.501_RunDisplacedJet2024G)
- [145.502](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.502_RunEGamma02024G)
- [145.503](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.503_RunHcalNZS2024G)
- [145.504](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.504_RunJetMET02024G)
- [145.505](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.505_RunMuon02024G)
- [145.506](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.506_RunMuonEG2024G)
- [145.507](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.507_RunNoBPTX2024G)
- [145.508](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.508_RunPark2MuonLowMass02024G)
- [145.509](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.509_RunParkingHH2024G)
- [4.22](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/4.22_RunCosmics2011A)
- [145.21](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.21_RunParkingLLP2024D)
- [136.902](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/136.902_RunDoubleMuon2016H)
- [136.903](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/136.903_RunDoubleMuon2017B)
- [136.901](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/136.901_RunDoubleMuon2016C)
- [1003.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/1003.0_RunMinBias2012A)
- [136.904](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/136.904_RunDoubleMuon2018C)
- [145.108](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.108_RunPark2MuonLowMass02024C)
- [145.2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.2_RunBTagMu2024D)
- [145.104](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.104_RunJetMET02024C)
- [145.105](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.105_RunMuon02024C)
- [140.56](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/140.56_RunHI2018)
- [145.107](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.107_RunNoBPTX2024C)
- [1001.2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/1001.2_RunZeroBias2017F)
- [145.101](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.101_RunDisplacedJet2024C)
- [145.102](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.102_RunEGamma02024C)
- [145.103](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.103_RunHcalNZS2024C)
- [145.214](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.214_RunZeroBias2024D)
- [145.6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.6_RunBTagMu2024H)
- [145.5](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.5_RunBTagMu2024G)
- [145.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.4_RunBTagMu2024F)
- [145.3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.3_RunBTagMu2024E)
- [145.211](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.211_RunParkingSingleMuon02024D)
- [145.212](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.212_RunParkingVBF02024D)
- [145.213](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/runTheMatrixINPUT-results/145.213_RunTau2024D)
</details>



## AddOn Tests

- [fastsim](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/fastsim)
```
A fatal system signal has occurred: abort signal
```
- [fastsim1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/fastsim1)
```
A fatal system signal has occurred: abort signal
```
- [fastsim2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/fastsim2)
```
A fatal system signal has occurred: abort signal
```
<details>
<summary>Expand to see more addon errors ...</summary>

- [hlt_mc_Fake](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_mc_Fake)
- [hlt_mc_Fake1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_mc_Fake1)
- [hlt_mc_Fake2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_mc_Fake2)
- [hlt_mc_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_mc_GRun)
- [hlt_mc_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_mc_GRun)
- [hlt_mc_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_mc_HIon)
- [hlt_mc_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_mc_HIon)
- [hlt_mc_PIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_mc_PIon)
- [hlt_mc_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_mc_PRef)
- [hlt_mc_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_mc_PRef)
- [hlt_mc_Special](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_mc_Special)
- [hlt_mc_Special](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_mc_Special)
- [hlt_data_Fake](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_data_Fake)
- [hlt_data_Fake1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_data_Fake1)
- [hlt_data_Fake2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_data_Fake2)
- [hlt_data_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_data_GRun)
- [hlt_data_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_data_GRun)
- [hlt_data_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_data_HIon)
- [hlt_data_PIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_data_PIon)
- [hlt_data_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_data_PRef)
- [hlt_data_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_data_PRef)
- [hlt_data_Special](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_data_Special)
- [hlt_data_Special](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/43910/addOnTests/hlt_data_Special)
</details>

jfernan2 commented:
@jhgoh I do not fully understand, according to your description, the error should not be present since #47134 has been already merged? Am I missing anything?

bsunanda commented:
There are 3 more PRs that are to be merged #47158, #47159 and #47160. In fact all tests have to be used from XML files and not from DB. The DB payloads need to be recreated once this PR is merged.
________________________________
From: jfernan2 ***@***.***>
Sent: 23 January 2025 00:27
To: cms-sw/cmssw ***@***.***>
Cc: Sunanda Banerjee ***@***.***>; Comment ***@***.***>
Subject: Re: [cms-sw/cmssw] Modify RPC isFront function to match criteria with will be modified RPC geometries for Run, Run 3 and Run 4 (PR #47138)


@jhgoh<https://github.com/jhgoh> I do not fully understand, according to your description, the error should not be present since #47134<https://github.com/cms-sw/cmssw/pull/47134> has been already merged? Am I missing anything?

—
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/47138#issuecomment-2608029474>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGMZORNIPGSDOBLT27O5632L7SZLAVCNFSM6AAAAABVPZ3DUSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMMBYGAZDSNBXGQ>.
You are receiving this because you commented.Message ID: ***@***.***>

jfernan2 commented:
> There are 3 more PRs that are to be merged #47158, #47159 and #47160. In fact all tests have to be used from XML files and not from DB. The DB payloads need to be recreated once this PR is merged.

But those are included in the test_parameters of this PR, I understand this is not enough either? Thanks

eigen1907 commented:
> > There are 3 more PRs that are to be merged #47158, #47159 and #47160. In fact all tests have to be used from XML files and not from DB. The DB payloads need to be recreated once this PR is merged.
> 
> But those are included in the test_parameters of this PR, I understand this is not enough either? Thanks

The three PRs change the scenario for calling xml geometry. 

The DB payload hasn't changed yet, so scenarios that call geometry from the DB will throw an error. 

Best regards,

jfernan2 commented:
Ok thanks. So, I understand this PR should be approved and merged to create the DB payloads but we cannot test it in advance. Is there any local test which can prove it works? Thanks

perrotta commented:
assign alca

cmsbuild commented:
New categories assigned: alca

@atpathak,@consuegs,@perrotta you have been requested to review this Pull request/Issue and eventually sign? Thanks

perrotta commented:
I don't think this PR cannot be approved now: we cannot modify Run2+Run3 geometries now without a thorough discussion.
Would it be limited to 2025 it would be a different story (but still the changes have to be discussed)

perrotta commented:
> assign alca

(at least until a modification of the conditions is required to have it working)

eigen1907 commented:
> Ok thanks. So, I understand this PR should be approved and merged to create the DB payloads but we cannot test it in advance. Is there any local test which can prove it works? Thanks

I'm attaching a slide with more background information and a local test (backup slide) manual for this PR.
[RPCGeometryChanges_250120_JShin](https://indico.cern.ch/event/1490623/contributions/6334824/attachments/3000178/5286762/RPCGeometryChanges_250120_JShin.pdf)

eigen1907 commented:
> I don't think this PR cannot be approved now: we cannot modify Run2+Run3 geometries now without a thorough discussion. Would it be limited to 2025 it would be a different story (but still the changes have to be discussed)

[#47158](https://github.com/cms-sw/cmssw/pull/47158), [#47159](https://github.com/cms-sw/cmssw/pull/47159) and [#47160](https://github.com/cms-sw/cmssw/pull/47160).

If so, it seems that these three PRs [#47158](https://github.com/cms-sw/cmssw/pull/47158), [#47159](https://github.com/cms-sw/cmssw/pull/47159) and [#47160](https://github.com/cms-sw/cmssw/pull/47160) should not be applied as well.
They only work without error when applied in conjunction with the current open [#47138](https://github.com/cms-sw/cmssw/pull/47138 .

bsunanda commented:
Before making payloads like to know which geometries are to be updated. I would like this to be discussed in PPD and/or SIM meeting

bsunanda commented:
@cmsbuild Please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 16KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/44140/summary.html
**COMMIT**: 185af904401064400d913724e9cf0316c675852a
**CMSSW**: CMSSW_15_0_X_2025-02-02-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/47138/44140/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [removed 5 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-02-02-2300+09a151/66727/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 9232 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-02-02-2300+09a151/66727/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 4016938
 - DQMHistoTests: [Total failures: 87807](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-02-02-2300+09a151/66727/validateJR.html)
 - DQMHistoTests: Total nulls: 81
 - DQMHistoTests: Total successes: 3929030
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 218 log files, 189 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2025-02-02-2300+09a151/66727/triggerResults): found differences in 4 / 48 workflows

bsunanda commented:
@jfernan2, @mandrenguyen, @perrotta Can you please sign this PR

cmsbuild commented:
Milestone for this pull request has been moved to CMSSW_15_1_X. Please open a backport if it should also go in to CMSSW_15_0_X.

civanch commented:
This PR was discussed at SIM meeting (https://indico.cern.ch/event/1507600/) and muon meeting (https://indico.cern.ch/event/1511920/). We agree that this PR is valid, and should be merged to 15_1. It is needed to disable isFront() check for RPC for the time being.

perrotta commented:
The suggestion proposed in the muon meeting was about implementing an `if-then-else` structure, either ruled by a modifier or by a check in the innermost RPC layer, so that the `isFront` check can be maintained and specialized to the old and updated geometries, respectively. Is there any plan to proceed as such?

jhgoh commented:
> The suggestion proposed in the muon meeting was about implementing an `if-then-else` structure, either ruled by a modifier or by a check in the innermost RPC layer, so that the `isFront` check can be maintained and specialized to the old and updated geometries, respectively. Is there any plan to proceed as such?

Thank you for your suggestions. One of the my concern was adding too much modifications in the base-classes regarding general geometry and related modules, only to add hard-coded exceptional buggy cases.
In any case, if we really want to fix this wrong-staggering only for late-2025 geometry but keep the past geometries as buggy as is, I think it is possible.

I'm writing some notes for @eigen1907 to let him get some idea:

The functions of `MuonRPCDetLayerGeometryBuilder` class are all static ones (see https://github.com/cms-sw/cmssw/blob/master/RecoMuon/DetLayers/src/MuonRPCDetLayerGeometryBuilder.h#L29-L43), so it is not a good idea to set era-dependancy flag as its member variable. 

The one you have to look at/modify is: `MuonRPCDetLayerGeometryBuilder::buildEndcapLayers(const RPCGeometry& geo)` which is called by `MuonDetLayerGeometryESProducer` (https://github.com/cms-sw/cmssw/blob/master/RecoMuon/DetLayers/plugins/MuonDetLayerGeometryESProducer.cc#L99). 
In the constructor of the `MuonDetLayerGeometryESProducer` a `ParameterSet` is passed, so that we can give era-dependent flag with the era-modifiers. Then you can pass additional flag to apply corrected even/odd rule in building the endcap layers. https://github.com/cms-sw/cmssw/blob/master/RecoMuon/DetLayers/plugins/MuonDetLayerGeometryESProducer.cc#L99

perrotta commented:
@jhgoh @eigen1907 the suggestion made at the muon meeting (not by me) of looking at the innermost ring and then use it to decide which kind of geometry order has to be considered for the rest of the RPC was also valid, and it does not require defining any era.

civanch commented:
Hi all, If the method isFront() is needed for reconstruction of other run time purpose, I would agree with last comment of @perrotta that internal check may be done and modifier is not needed. However, my understanding (may be wrong) is that this method was design for internal checks after geometry construction and is not used run time. If it is true, let us merge this PR and resolve RPC geometry problems - we are limited in time and the best to me is for the time being to disable these checks. It is possible to create an issue for better implementation of isFront() method and do that later.

jhgoh committed:
recover the font/back scheme of iRPCs, re-activate selftest function but rewriting the routine to check z-ordering separating by subsystems

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-47138/43663

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-47138/43663/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-47138/43663/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

jhgoh committed:
fix code-format

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-47138/43668

cmsbuild commented:
Pull request #47138 was updated. @atpathak, @cmsbuild, @consuegs, @jfernan2, @mandrenguyen, @perrotta can you please check and sign again.

bsunanda commented:
@cmsbuild Please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 16KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/44377/summary.html
**COMMIT**: 03c77014a6b445eb3a83dc6e0c34edc1c0c8a366
**CMSSW**: CMSSW_15_1_X_2025-02-13-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/47138/44377/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 2 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-02-13-1100+09a151/66909/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 2154 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-02-13-1100+09a151/66909/validateJR.html)
 - DQMHistoTests: Total files compared: 49
 - DQMHistoTests: Total histograms compared: 3905420
 - DQMHistoTests: [Total failures: 630](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-02-13-1100+09a151/66909/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3904770
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 48 files compared)
 - Checked 214 log files, 184 edm output root files, 49 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-02-13-1100+09a151/66909/triggerResults): found differences in 2 / 47 workflows

eigen1907 commented:
@perrotta Yes, this PR should be applied to match the newly implemented RPC geometry for this 15X release.

But it's been a while, so I'll retest locally and let you know as soon as it passes.

eigen1907 commented:
@perrotta I tested it locally again and found no issues. Could you please merge this PR?

perrotta commented:
@eigen1907 as far as I can see, `selfTest()` is still [deactivated](https://github.com/cms-sw/cmssw/blob/master/RecoMuon/DetLayers/src/MuRingForwardDoubleLayer.cc#L48) in  RecoMuon/DetLayers/src/MuRingForwardDoubleLayer.cc, after #47230. Shouldn't we reactivate it, in order to properly test this PR?

cmsbuild commented:
Milestone for this pull request has been moved to CMSSW_16_0_X. Please open a backport if it should also go in to CMSSW_15_1_X.

eigen1907 commented:
@perrotta After reactivating selfTest() locally and running the test, it threw errors for all global tags using the unupdated RPC geometries, exactly as predicted in this thread's discussion.   

However, the updated RPC geometries and the correctly updated isFront function to match them are clearly needed.

But actually i think it seems like an enormous amount of work to apply the updated RPC geometries to all the global tags that we were using.

Therefore, as also discussed in this thread, I am testing changing the `isFront` function to behave differently depending on the era (with Modifier), and it is being successfully completed.
(local branch under testing: https://github.com/cms-sw/cmssw/commit/e90a60d8ada986e5a3037a352f6f221abe8a144c)

But it seems necessary to review whether this is the correct behavior from a software architecture viewpoint.

I'm also not sure how to handle the PR.  Could you please review and comment on this?

eigen1907 committed:
Add an RPC Modifier class for Run3 2025 and add it to the ModifierChain within the Run3 2025 Era

eigen1907 committed:
Add an RPC Modifier class for Phase2 and add it to the ModifierChain within the Phase2 Era

eigen1907 committed:
Modified MuonRPCDetLayerGeometryBuilder::isFront() to use a different algorithm based on the useUpdatedRPCIsFront static variable
- useUpdatedRPCIsFront is defined internally within MuonDetLayerGeometryESProducer
- useUpdatedRPCIsFront is set according to Era using Modifier in muonDetLayerGeometry_cfi.py
    - The legacy isFront algorithm operates until 2025
    - The updated isFront algorithm operates from 2025

eigen1907 committed:
Reactivated selfTest following updates to RPC geometry and the isFront function

eigen1907 committed:
fix code-format

perrotta reviewed (commented):
It looks like this PR is still a work in progress: please advise when it is ready for review

eigen1907 committed:
Apply suggestions from code review

Co-authored-by: Andrea Perrotta <perrotta@cern.ch>

eigen1907 committed:
Apply suggestion from code review by perrotta

perrotta commented:
So, the tests allowed us to verify where there are issues with the Global Tags. Let remind here that up to now only 151X 2025 MC GTs were updated with the fixed RPC geometry, at least among the ones listed in Configuration/AlCa/python/autoCond.py

According to the logs I was able to inspect, I see two kinds of issues to be addressed:
- The fixed RPC geometry must be imported in the `150X_dataRun3_*` GTs in autoCond. To do so, the new GTs with the fixed RPC geometry must be created in 151X, analogously to what was done for the 2025 MC GTs in 151X
- As far as I understand, the HLT AddOn test for MC are currently run (see e.g. [1]) with `--era Run3_2024`, but with 2025 MC GTs as for `--conditions auto:run3_mc_GRun` (which corresponds to 'phase1_2025_realistic')

The first item should be investigated together with the Condition Preparation group (@arunhep @atpathak, and I can also help)

About the second item: @mmusich @Martin-Grunewald is there any reason for not moving to `--era Run3_2025` for those addOn tests? Or, if you really want to stick on a 2024 era, shouldn't you base the run3_mc_GRun GT in autoCond[HLT.py](https://github.com/cms-sw/cmssw/blob/8226eddd2c6a54ae97481ab24ab4a149dec9a665/Configuration/HLT/python/autoCondHLT.py#L39) to  'phase1_2024_realistic' GT instead?

[1] https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-09a151/48084/addOnTests/hlt_mc_GRun/cmdLog

mmusich commented:
> is there any reason for not moving to --era Run3_2025 for those addOn tests?

See https://github.com/cms-sw/cmssw/pull/48903

perrotta commented:
> > is there any reason for not moving to --era Run3_2025 for those addOn tests?
> 
> See #48903

Great, thank you @mmusich for pointing it out! We should have waited only one IB, then...
That PR is already merged in master: let retry the tests, then, to verify whether only 2025 data workflows keep failing.

perrotta commented:
please test

mmusich commented:
> let retry the tests, then, to verify whether only 2025 data workflows keep failing.

Just a note, the HLT recognizes no eras. If this is meant as a permanent change the HLT menus will need to be updated accordingly via CMSHLT ticket and regular integration procedures .

perrotta commented:
-1
- Just to keep this PR out of our radars
- We (alca) will get back to it as soon as anyone will propose  a working version of it.

cmsbuild commented:
Milestone for this pull request has been moved to CMSSW_20_0_X. Please open a backport if it should also go in to CMSSW_17_0_X.

makortel commented:
Just checking, is this PR intended for Phase 2 (20_1_X) or Run 3 legacy (17_0_X) or both?

cmsbuild commented:
Milestone for this pull request has been moved to CMSSW_20_1_X. Please open a backport if it should also go in to CMSSW_20_0_X.
